app/user/page.tsx

```diff
@@ -30,10 +52,41 @@ export default function AccountPage() {
   });
 
   useEffect(() => {
-    if (hatData) {
-      setHatId(hatData.toString());
+    if (rawHatData) {
+      const typedHatData: HatData = {
+        details: rawHatData[0],
+        maxSupply: BigInt(rawHatData[1]),
+        supply: BigInt(rawHatData[2]),
+        eligibility: rawHatData[3],
+        toggle: rawHatData[4],
+        imageUri: rawHatData[5],
+        numChildren: BigInt(rawHatData[6]),
+        mutable: rawHatData[7],
+        active: rawHatData[8],
+      };
+
+      setHatData(typedHatData);
+      fetchHatMetadata(typedHatData.details);
+    }
+  }, [rawHatData]);
+
+  const fetchHatMetadata = async (detailsUrl: string) => {
```

Comment:
For the future, we should put this into its own file so it's nicely encapsulated.